### ISSUE TITLE
Figuring out downgrade performance at v1.1.1

### ISSUE THREAD

**Issue by sifnoc** [summa-dev/summa-solvency#258]
We conducted a benchmark on the latest master commit (d6e85c9affebef338846d5fb13e40eb8ab657811) and observed a decrease in performance compared to the previous benchmark results from `merkle_sum_tree_v1.1` (b289167eee67098cb4cac79849caece5f8978939). Note that we did not perform the benchmark on `merkle_sum_tree_v1.1.1` (b832e5e2814a7222576a15a8008685983f1beffd).

For your instances, I will share benchmark results.
 
Below are the benchmark logs for the latest master. To observe the performance changes, I ran the benchmark for `merkle_sum_tree_v1.1.1` just before running the latest master:

```
Benchmarking generate zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit: Analyzing
generate zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit
                        time:   [418.87 ms 421.57 ms 424.66 ms]
                        change: [-5.1815% -3.7495% -2.5007%] (p = 0.00 < 0.05)
                        Performance has improved.

Start:   None Creating params
End:     None Creating params ......................................................170.017ms
Start:   Creating proof
End:     Creating proof ............................................................418.963ms
proof size in bytes: 1632
Benchmarking verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit
Benchmarking verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit: Warming up for 3.0000 s
Benchmarking verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit: Collecting 10 samples in estimated 5.0644 s (1265 iterations)
Benchmarking verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit: Analyzing
verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit
                        time:   [3.9985 ms 3.9999 ms 4.0016 ms]
                        change: [-7.7839% -6.9809% -6.2576%] (p = 0.00 < 0.05)
                        Performance has improved.
```

As a result, we can clearly observe that performance has improved in the latest master compared to `merkle_sum_tree_v1.1.1`.

Now, let's examine the benchmark results for `merkle_sum_tree_v1.1.1`. For comparison, I conducted the `merkle_sum_tree_v1.1` benchmark immediately prior to running v1.1.1:

```
Benchmarking generate zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit: Analyzing
generate zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit
                        time:   [395.77 ms 398.18 ms 400.75 ms]
                        change: [+2.9267% +4.1539% +5.3029%] (p = 0.00 < 0.05)
                        Performance has regressed.

Start:   None Creating params
End:     None Creating params ......................................................151.553ms
Start:   Creating proof
End:     Creating proof ............................................................396.689ms
proof size in bytes: 1632
Benchmarking verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit
Benchmarking verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit: Warming up for 3.0000 s
Benchmarking verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit: Collecting 10 samples in estimated 5.0566 s (1265 iterations)
Benchmarking verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit: Analyzing
verify zk proof - tree of 2 power of 20 entries with 1 currencies mst inclusion circuit
                        time:   [3.9955 ms 3.9992 ms 4.0007 ms]
                        change: [+7.1392% +7.5718% +8.0147%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 1 outliers among 10 measurements (10.00%)
```

In conclusion, the performance comparison yields the following sequence:
v1.1 > latest master > v1.1.1

To address the performance this regression issues, we have to review the changes made between these versions. Please refer to the comparative changes at: 
https://github.com/summa-dev/summa-solvency/compare/merkle_sum_tree_v1.1...merkle_sum_tree_v1.1.1
